src/Sql/Sql/Elastic Jobs/Cmdlet/JobExecution/StopAzureSqlElasticJob.cs

```diff
@@ -0,0 +1,171 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Sql.ElasticJobs.Model;
+using Microsoft.Azure.Commands.Sql.ElasticJobs.Cmdlet.JobExecution;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Sql.ElasticJobs.Cmdlet
+{
+    /// <summary>
+    /// Defines the Stop-AzureRmSqlElasticJob Cmdlet
+    /// </summary>
+    [Cmdlet(VerbsLifecycle.Stop, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlElasticJob",
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = DefaultParameterSet)]
+    [OutputType(typeof(AzureSqlElasticJobExecutionModel))]
+    public class StopAzureSqlElasticJob : AzureSqlElasticJobExecutionCmdletBase<AzureSqlElasticJobExecutionModel>
+    {
+        /// <summary>
+        /// Gets or sets the resource group name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 0,
+            HelpMessage = "The resource group name")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Agent's Control Database Object
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The Agent Control Database Object")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlElasticJobExecutionModel ParentObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Agent's Control Database Object
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The job execution resource id")]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
```

Comment:
@johnpaulkee same comment about `-ResourceId`

Reply:
As discussed we'll address the Resource Id changes in a separate PR to reduce complexity of this one. We can do a one-off release with these changes and then add -ResourceId parameter  later.